docs/guide/src/en/contributing/walkthrough_readme.md

```diff
@@ -63,64 +63,74 @@ pub struct VirtualDom {
 ```
 
 > What is a [slab](https://docs.rs/slab/latest/slab/)?
+>
 > A slab acts like a hashmap with integer keys if you don't care about the value of the keys. It is internally backed by a dense vector which makes it more efficient than a hashmap. When you insert a value into a slab, it returns an integer key that you can use to retrieve the value later.
 
 > How does Dioxus use slabs?
-> Dioxus uses "synchronized slabs" to communicate between the renderer and the VDOM. When an node is created in the Virtual Dom, a ElementId is passed along with the mutation to the renderer to identify the node. These ids are used by the Virtual Dom to reference that nodes in future mutations like setting an attribute on a node or removing a node.
-> When the renderer sends an event to the Virtual Dom, it sends the ElementId of the node that the event was triggered on. The Virtual Dom uses this id to find the node in the slab and then run the necessary event handlers.
+>
+> Dioxus uses "synchronized slabs" to communicate between the renderer and the VDOM. When a node is created in the Virtual DOM, an (elementId, mutation) pair is passed to the renderer to identify that node, which the renderer will then render in actual DOM. These ids are also used by the Virtual Dom to reference that node in future mutations, like setting an attribute on a node or removing a node. When the renderer sends an event to the Virtual Dom, it sends the ElementId of the node that the event was triggered on. The Virtual DOM uses this id to find that node in the slab and then run the necessary event handlers.
```

Comment:
I don't know what this means exactly here and didn't feel like I could glean enough context to reword it well myself.

Reply:
Mutations are the way dioxus communicates what should be rendered. When the virtual DOM creates an element, it generates a mutation with an unique id for the element. The id can both:
1. Identify the element in the virtual DOM. This is used to identify where to trigger event listeners when the renderer sends an event back to the virtual DOM. ([desktop renderer events with element id](https://github.com/DioxusLabs/dioxus/blob/master/packages/interpreter/src/interpreter.js#L461))
2. Identify the element in the renderer. This is used to modify or remove the element later.

I called it a "synchronized slab" because:
- The virtual DOM has the original slab. New elements are created and removed in that slab.
- The renderer has a mirrored slab that changes with mutations to reflect the original slab but with platform-specific elements instead of virtual nodes. In the desktop renderer, this is the [nodes](https://github.com/DioxusLabs/dioxus/blob/master/packages/interpreter/src/interpreter.js#L69) variable. Elements being created, [sets an element at the index of the element id](https://github.com/DioxusLabs/dioxus/blob/master/packages/interpreter/src/interpreter.js#L131). Elements will be reused when the virtual DOM removes the node id and then assigns the element id to a new element (overriding the previous element).